addons/website_blog/models/website_blog.py

```diff
@@ -254,6 +257,9 @@ def write(self, cr, uid, ids, vals, context=None):
             ids = [ids]
         if 'content' in vals:
             vals['content'] = self._postproces_content(cr, uid, ids[0], vals['content'], context=context)
+        post = self.browse(cr, uid, ids, context)
+        if 'website_published' in vals and vals['website_published'] and post.create_date == post.date_published:
+            vals['date_published'] = datetime.now()
```

Comment:
we have introduce new field `date_published` then what the purpose of compare create_date with date_published str ?